File: SchoolManagementApi/DTOs/Subjects/SubjectResponse.Dto.cs
namespace SchoolManagementApi.DTOs.Subjects;

public class SubjectResponseDto
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public int Credits { get; set; }
    public int TeacherId { get; set; }

    public string TeacherName { get; set; } = null!;
}

File: SchoolManagementApiNode/src/subjects/dto/subject-response.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class SubjectResponseDto {
  @ApiProperty()
  id: number;

  @ApiProperty()
  name: string;

  @ApiProperty({ nullable: true })
  description: string | null;

  @ApiProperty()
  credits: number;

  @ApiProperty()
  teacherId: number;

  @ApiProperty()
  teacherName: string;
}
